rust/xaynet-server/src/state_machine/phases/unmask.rs

```diff
@@ -130,36 +136,44 @@ where
         model_agg
             .validate_unmasking(&mask)
             .map_err(UnmaskStateError::from)?;
+        self.shared.state.global_model = Some(Arc::new(model_agg.unmask(mask)));
 
-        Ok(model_agg.unmask(mask))
-    }
-
-    #[cfg(feature = "model-persistence")]
-    async fn save_global_model(&mut self, global_model: &Model) -> Result<(), UnmaskStateError> {
-        use tracing::warn;
-
-        let round_seed = &self.shared.state.round_params.seed;
-        let global_model_id = self
-            .shared
-            .store
-            .set_global_model(self.shared.state.round_id, &round_seed, global_model)
-            .await
-            .map_err(UnmaskStateError::SaveGlobalModel)?;
-        let _ = self
-            .shared
-            .store
-            .set_latest_global_model_id(&global_model_id)
-            .await
-            .map_err(|err| warn!("failed to update latest global model id: {}", err));
         Ok(())
     }
 }
 
-impl<S> PhaseState<Unmask, S>
+impl<T> PhaseState<Unmask, T>
 where
-    Self: Phase<S>,
-    S: Storage,
+    T: Storage,
 {
+    #[cfg(feature = "model-persistence")]
+    async fn save_global_model(&mut self) -> Result<(), UnmaskStateError> {
+        if let Some(ref global_model) = self.shared.state.global_model {
```

Comment:
here I would also use `unwrap` to avoid the `if let`
not sure if this is possible though

Reply:
i'll change it to an `expect()` as well. it should be possible because the `else` branch has return type `!` anyways.